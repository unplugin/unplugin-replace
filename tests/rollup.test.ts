import path from 'node:path'
import { describe, expect, test } from 'vitest'
import { rollupBuild } from '@vue-macros/test-utils'
import UnpluginReplace from '../src/rollup'

describe('rollup', () => {
  test('find string', async () => {
    const result = await rollupBuild(
      path.resolve(__dirname, 'fixtures/main.js'),
      [
        UnpluginReplace({
          'process.platform': '"darwin"',
        }),
      ],
    )
    expect(result).toMatchSnapshot()
    expect(result).not.contains('process.platform')
    expect(result).contains('"darwin"')
  })

  test('find regexp', async () => {
    const result = await rollupBuild(
      path.resolve(__dirname, 'fixtures/main.js'),
      [
        UnpluginReplace({
          values: [
            { find: /process\.\w+/g, replacement: 'null' },
            {
              find: /[A-Z]/,
              replacement: (id, match) => match[0].toLowerCase(),
            },
            {
              find: /console\.\w+/g,
              replacement: () => `console.info`,
            },
          ],
        }),
      ],
    )
    expect(result).toMatchSnapshot()
    expect(result).contains('console.info(dEV)')
    expect(result).contains(`console.info('hello null', platform)`)
  })
})
