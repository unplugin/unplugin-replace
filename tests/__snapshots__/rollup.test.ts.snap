// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup > find regexp 1`] = `
"// main.js
import 'node:process';

const platform = null;

console.info('hello null', platform);
console.info(dEV);

export { platform };
"
`;

exports[`rollup > find string 1`] = `
"// main.js
import 'node:process';

const platform = "darwin";

console.log('hello "darwin"', platform);
console.log(DEV);

export { platform };
"
`;
